bg_atlasgen/validate_atlases.py

```diff
@@ -0,0 +1,195 @@
+    _assert_close(y_min_mesh, y_min_scaled, resolution[1])
+    _assert_close(y_max_mesh, y_max_scaled, resolution[1])
+    _assert_close(x_min_mesh, x_min_scaled, resolution[2])
+    _assert_close(x_max_mesh, x_max_scaled, resolution[2])
+
+    return True
+
+
+def open_for_visual_check():
+    pass
+
+
+def validate_checksum():
+    pass
+
+
+def check_additional_references():
+    # check additional references are different, but have same dimensions
+    pass
+
+
+def validate_atlas(atlas_name, version, all_validation_functions):
+    """Validates the latest version of a given atlas"""
+
+    print(atlas_name, version)
+    BrainGlobeAtlas(atlas_name)
+    updated = get_atlases_lastversions()[atlas_name]["updated"]
+    if not updated:
+        update_atlas(atlas_name)
+    Path(get_brainglobe_dir()) / f"{atlas_name}_v{version}"
+
+    validation_function_parameters = [
+        # validate_atlas_files(atlas_path: Path)
+        (Path(get_brainglobe_dir() / f"{atlas_name}_v{version}"),),
+        # validate_mesh_matches_image_extents(atlas: BrainGlobeAtlas)
+        (BrainGlobeAtlas(atlas_name),),
+        # open_for_visual_check()
+        (),
+        # validate_checksum()
+        (),
+        # check_additional_references()
+        (),
+        # validate_atlas(atlas_name, version)
+        (atlas_name, version),
+    ]
+
+    # list to store the errors of the failed validations
+    failed_validations = []
+    successful_validations = []
+
+    for i, validation_function in enumerate(all_validation_functions):
+        try:
+            validation_function(*validation_function_parameters[i])
+            successful_validations.append((atlas_name, validation_function))
+        except AssertionError as error:
+            failed_validations.append((atlas_name, validation_function, error))
+
+    return successful_validations, failed_validations
+
+
+def validate_mesh_structure_pairs(atlas_path: Path):
+    json_path = Path(atlas_path / "structures.json")
+    obj_path = Path(atlas_path / "meshes")
+
+    with open(json_path, "r") as file:
+        json_file = json.load(file)
+
+    obj_file_list = [
+        file for file in os.listdir(obj_path) if file.endswith(".obj")
+    ]
+    target_key = "id"
+    id_numbers = [item[target_key] for item in json_file if target_key in item]
+
+    matching_files = [
+        f"{num}.obj" for num in id_numbers if f"{num}.obj" in obj_file_list
+    ]
+    missing_files = [
+        num for num in id_numbers if f"{num}.obj" not in obj_file_list
+    ]
+
+    print(f"IDs without corresponding obj files: {missing_files}")
+    print(f"IDs with corresponding obj files: {matching_files}")
+
+
+if __name__ == "__main__":
+    # list to store the validation functions
+    all_validation_functions = [
+        validate_atlas_files,
+        validate_mesh_matches_image_extents,
+        open_for_visual_check,
+        validate_checksum,
+        check_additional_references,
```

Comment:
```suggestion
        check_additional_references,
        validate_mesh_structure_pairs,
```
We need to add the validation function to the list of validation functions so it actually gets run? We'll also need to pass arguments for it in the `validate_atlas()` function.